aten/src/THC/THCApply.cuh

```diff
@@ -141,22 +143,21 @@ template <typename Op,
 __launch_bounds__(THC_APPLY_THREADS_PER_BLOCK, THC_APPLY_BLOCKS_PER_SM)
 #endif
 __global__ void
-kernelPointwiseApply2(TensorInfo<Ta, IndexType> a,
-                      TensorInfo<Tb, IndexType> b,
+kernelPointwiseApply2(const OffsetInfo<Ta, IndexType, ADims> a,
+                      const OffsetInfo<Tb, IndexType, BDims> b,
                       IndexType totalElements,
                       Op op) {
-  for (IndexType linearIndex = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
keeping these old for loops, as you found out, cast blockIdx.x to IndexType etc. to avoid overflow

Reply:
Done.